src/computation/logic/distribute-levelling-seats.ts

```diff
@@ -90,6 +65,164 @@ export function distributeLevelingSeats(
         );
     }
 
-    const levelingSeatDistribution = dictionaryToArray(partyRestQuotients);
-    return levelingSeatDistribution;
+    const levelingSeatDistributionArray = dictionaryToArray(partyRestQuotients);
+    return levelingSeatDistributionArray;
+}
+
+/**
+ * Filters out parties that do not receive more seats in the national distribution than the district distribution.
+ * It returns the remaining party codes and the final national distribution of seats.
+ *
+ * @param levelingPartyCodes The party codes to filter
+ * @param payload The computation payload
+ * @param partyResults The party results for each party
+ */
+function nationalDistributionFilter(
+    levelingPartyCodes: string[],
+    payload: ComputationPayload,
+    partyResults: Dictionary<PartyResult>
+): { nationalDistribution: DistributionResult; levelingPartyCodes: string[] } {
```

Comment:
Shouldn't this be typed more explicitly?

Reply:
Fixed.